packages/api/src/hooks/useSubscribedBalance.ts

```diff
@@ -0,0 +1,38 @@
+import { useEffect } from 'react';
+import useAuthorize from './useAuthorize';
+import useSubscription from '../useSubscription';
+import { displayMoney } from '../utils';
+import useActiveWalletAccount from './useActiveWalletAccount';
+
+/**
+ * Hook to display live, subscribed balance.
+ * Use when you have only one wallet on screen.
+ */
+const useSubscribedBalance = () => {
+    const { data: activeWallet } = useActiveWalletAccount();
+    const { data: account } = useAuthorize();
+
+    const { subscribe, data: balanceData, unsubscribe, ...rest } = useSubscription('balance');
```

Comment:
UseBalance is already being refetched every 30 seconds 🤔

Reply:
yes, and that hook is specifically added to have live subscription, 
so when you e.g. deposit FIAT or reset balance, its immediately visible, 

if its not imediately visible, its basically looks broken to end user,

Reply:
@wojciech-deriv I believe you can just simply call `invalidate('balance')` on those actions.

Reply:
I agree with @farzinmirzaie. We are already hitting rate limit for balance sometimes. 😿

Reply:
Does backend guarantee that balance is updated before any of the operations like deposit, balance reset, withdrawal is confirmed (and for every other user-initiated operation which impacts balance)?
Otherwise, we can have race hazards and we can invalidate before update. In my mind, I did not wanted to make any unexplicit assumptions about how backend works (especially as I know we already had quite a couple race hazards in past and especially given that I would expect some renovations in that area).

Reply:
The preferred approach, I would imagine, should be to subscribe to the balance endpoint so we get notified immediately when the balance changes, instead of invalidating the react-query cache. It feels "side-effecty" to me, and the polling feature seems like a work around?

Reply:
Also, after some investigation, realised that "invalidate" will trigger balance fetching for all account, meaning, in current implementation, it will trigger authorise request for all accounts (e.g. 5 wallets) and balance request for all wallets, so extra 10 request for client with five wallets. While subscription, won't trigger any new request, and will avoid all the weird logic of reauthorisation. So arguable, subscribe is way less likely to trigger any backend limits than invalidate.

Reply:
(assuming that backend limits are not triggered by the fact that we're sharing accounts internally within team and PMs, which might actually be the case given dev teams and product teams do use the same ones)